ownership-chain/runtime/laos/src/configs/fee.rs

```diff
@@ -0,0 +1,201 @@
+		ConsensusEngineId,
+	};
+	use sp_runtime::{
+		codec::{Decode, Encode},
+		testing::Header,
+		traits::Header as HeaderT,
+		BuildStorage,
+	};
+
+	type Block = frame_system::mocking::MockBlock<Test>;
+
+	// Configure a mock runtime to test the pallet.
+	frame_support::construct_runtime!(
+		pub enum Test
+		{
+			System: frame_system,
+			Authorship: pallet_authorship,
+			Balances: pallet_balances,
+		}
+	);
+
+	type AccountId = u64;
+	type Balance = u128;
+
+	#[derive_impl(frame_system::config_preludes::TestDefaultConfig as frame_system::DefaultConfig)]
+	impl frame_system::Config for Test {
+		type Block = Block;
+		type AccountData = pallet_balances::AccountData<Balance>;
+	}
+
+	parameter_types! {
+		pub const MaxTokenUriLength: u32 = 512;
+		pub const ExistentialDeposit: u128 = 1;
+	}
+
+	impl pallet_balances::Config for Test {
+		type MaxReserves = ();
+		type ReserveIdentifier = [u8; 4];
+		type MaxLocks = ();
+		type Balance = Balance;
+		type RuntimeEvent = RuntimeEvent;
+		type DustRemoval = ();
+		type ExistentialDeposit = ExistentialDeposit;
+		type AccountStore = System;
+		type WeightInfo = ();
+		type RuntimeHoldReason = ();
+		type FreezeIdentifier = ();
+		type MaxHolds = ();
+		type MaxFreezes = ();
+	}
+	impl pallet_authorship::Config for Test {
+		type FindAuthor = AuthorGiven;
+		type EventHandler = ();
+	}
+
+	const TEST_ID: ConsensusEngineId = [1, 2, 3, 4];
+
+	pub struct AuthorGiven;
+
+	impl frame_support::traits::FindAuthor<u64> for AuthorGiven {
+		fn find_author<'a, I>(digests: I) -> Option<u64>
+		where
+			I: 'a + IntoIterator<Item = (ConsensusEngineId, &'a [u8])>,
+		{
+			for (id, mut data) in digests {
+				if id == TEST_ID {
+					return u64::decode(&mut data).ok()
+				}
+			}
+
+			None
+		}
+	}
+
+	fn seal_header(mut header: Header, author: u64) -> Header {
+		{
+			let digest = header.digest_mut();
+			digest.logs.push(DigestItem::PreRuntime(TEST_ID, author.encode()));
+			digest.logs.push(DigestItem::Seal(TEST_ID, author.encode()));
+		}
+
+		header
+	}
+
+	fn create_header(number: u64, parent_hash: H256, state_root: H256) -> Header {
+		Header::new(number, Default::default(), state_root, parent_hash, Default::default())
+	}
+
+	fn new_test_ext() -> sp_io::TestExternalities {
+		let t = frame_system::GenesisConfig::<Test>::default().build_storage().unwrap();
+		t.into()
+	}
+
+	fn initialize_block_and_set_author(number: u64, author: AccountId) {
+		let mut header =
+			seal_header(create_header(number, Default::default(), [1; 32].into()), author);
+
+		header.digest_mut().pop(); // pop the seal off.
+		System::reset_events();
+		System::initialize(&number, &Default::default(), header.digest());
+	}
+
+	#[test]
+	fn sets_author_lazily() {
+		new_test_ext().execute_with(|| {
+			let author = 42;
+
+			initialize_block_and_set_author(1, author);
+			assert_eq!(Authorship::author(), Some(author));
+		});
+	}
+
+	#[test]
+	fn all_fee_should_go_to_block_author() {
+		new_test_ext().execute_with(|| {
```

Comment:
could you consider to check that `total_issuance` stays the same as you did in the following test?
https://github.com/freeverseio/laos/blob/137d88b8254ded2c5429b598a477351240fd5f42/ownership-chain/runtime/laos/src/configs/fee.rs#L195-L196